lib/ruby_lsp/ruby_lsp_rails/code_lens.rb

```diff
@@ -112,6 +119,50 @@ def on_class_node_leave(node)
 
       private
 
+      sig { params(node: Prism::ClassNode).void }
+      def add_route_code_lenses_to_actions(node)
+        public_method_nodes = T.must(node.body).child_nodes.take_while do |node|
+          !node.is_a?(Prism::CallNode) || ![:protected, :private].include?(node.name)
+        end
+        public_method_nodes.grep(Prism::DefNode)
+        public_method_nodes.each do |public_method_node|
+          public_method_node = T.cast(public_method_node, Prism::DefNode)
+          add_route_code_lense_to_action(public_method_node, class_node: node)
+        end
+      end
+
+      sig { params(node: Prism::DefNode, class_node: Prism::ClassNode).void }
+      def add_route_code_lense_to_action(node, class_node:)
+        route = @client.route(
+          controller: class_node.constant_path.slice,
+          action: node.name.to_s,
+        )
+
+        if route
+          path = route.dig(:path)
+          verb = route.dig(:verb)
+          source_location = route.dig(:source_location)
+
+          arguments = [
+            source_location,
+            {
+              start_line: node.location.start_line - 1,
+              start_column: node.location.start_column,
+              end_line: node.location.end_line - 1,
+              end_column: node.location.end_column,
+            },
+          ]
+
+          @_response << create_code_lens(
+            node,
+            title: [verb, path].join(" "),
+            command_name: "rubyLsp.openFile",
```

Comment:
It seems kind of silly to me that there's no native support in editors for a file opening command (that I can see). Is there a way around this, or should I proceed with my PR that implements this over on the extension?

Reply:
I had a patch like this coming this week. I don't work for Shopify and have been really busy at work so I didn't have time to get the tests done yet, so I wasn't pushing it up.

The format I had for routes was:

            name: route.name,
            verb: route.verb,
            path: route.path.sub(/...format./, ""),
            controller: route.controller,
            action: route.action,
            source_location: "#{::Rails.application.root}/{route.source_location.sub(":", "#L")}",
            views: ActiveSupport::Inflector.safe_constantize((route.controller.to_s + "_controller").camelize).new.lookup_context.find_all("#{route.controller}/#{route.action}").map do |route|
                     { filename: route.identifier, format: route.format }
                   end,
          }

If you use vscode.open, you can do this for VS Code at least.  

If you want to do it all you could use just the uri for non vscode editors (but I haven't checked on how to do this yet).  

My lens looked like: 
![Screenshot 2024-02-02 at 10 46 00 PM](https://github.com/Shopify/ruby-lsp-rails/assets/443111/95e2128e-aa64-47fc-a5d0-1618537771a4)

Reply:
Can you please include the views as I have code action coming to open corresponding views?

Reply:
Also note, if you use #L#{line_number} you can jump to the right line.

Reply:
Either way I will be doing the views.  I am trying to replicate most of Rubymine for my team so I will probably do an add-on if needed.

Reply:
Thanks for the co-authorship :)

Reply:
From looking at the [spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_codeLens), a Code Lens doesn't _need_ to have a command associated with it, so we could ship this with only the label, and deal with the command behaviour in a follow-up PR.

Reply:
@rickcaudill thanks for those suggestions. I think we can handle it separately, so I've created https://github.com/Shopify/ruby-lsp-rails/issues/384 to track.